Update the migration module guide (#1399)
memgraph/documentation: pages/data-migration/migrate-from-neo4j-using-single-cypher-query.mdx

docker run -it --rm -p 7688:7687 memgraph/memgraph-mage
```

> ## Setup migration indices

Reviewer: ```suggestion
## Create migration indices
```

Before we do the magic Cypher command, we need to create 2 necessary indices in order to speed up the migration process:

```cypher
CREATE INDEX ON :__MigrationNode__;
CREATE INDEX ON :__MigrationNode__(__elementId__);
```

We explain the necessity of these indices in the following paragraph.

> ## The magic Cypher command for seamless migration

Reviewer: ```suggestion
## Run the migration query
```

To migrate the entire graph from Neo4j to Memgraph, use the following Cypher query in Memgraph:

```cypher
CALL migrate.neo4j(
  "MATCH (n)-[r]->(m) RETURN labels(n) AS src_labels, type(r) as rel_type, labels(m) AS dest_labels, elementId(n) AS src_id, elementId(m) AS dest_id, properties(n) AS src_props, properties(r) AS edge_props, properties(m) AS dest_props", 
  {host: "localhost", port: 7687})
YIELD row 
MERGE (n:__MigrationNode__ {__elementId__: row.src_id})
MERGE (m:__MigrationNode__ {__elementId__: row.dest_id})
SET n:row.src_labels
SET m:row.dest_labels
SET n += row.src_props
SET m += row.dest_props
CREATE (n)-[r:row.rel_type]->(m)
SET r += row.edge_props;
```

- The query makes sure that all triplets are migrated from Neo4j to Memgraph.

> - Configuration object is the second argument in the query which is establishing a driver connection to Neo4j

Reviewer: ```suggestion
- The second argument in the query is the configuration object, (`host` and `port`) which is establishing a driver connection to Neo4j.
```

> - Since nodes can have an arbitrary number of labels, it is therefore necessary to ensure a single index is utilized
> while performing the `MERGE` command. Furthermore, Neo4j has a built-in `elementId(node)` function which acts as a global ID
> for the merge command to successfully transition the correct set of nodes into Memgraph.

Reviewer: ```suggestion
- Because nodes in Neo4j can have multiple labels, we need a reliable way to ensure a single index is utilized during the `MERGE` command. To achieve this, we use a single `__MigrationNode__ index`. Neo4j has a built-in `elementId(node)` function which acts as a global ID
for the merge command to successfully transition the correct set of nodes into Memgraph.
```

> - After we have correctly merged the nodes, we can then dinamically assign the labels with the `:row.src_labels` and `:row.dest_labels` constructs.
> - Relationship creation does not need a `MERGE` statement, since the cardinality of all triplets is in fact the cardinality of relationships in the graph.
> - Relationship type is a single string which is dinamically transported using the `:row.rel_type` constructs
> - Relationship properties are also added at the end of the query

Reviewer: ```suggestion
- After we have correctly merged the nodes, we can then dinamically assign the labels with the `:row.src_labels` and `:row.dest_labels` constructs.
- Relationship creation does not need a `MERGE` statement, since the cardinality of all triplets is in fact the cardinality of relationships in the graph.
- Relationship type is a single string which is dinamically transported using the `:row.rel_type` constructs.
- Relationship properties are also added at the end of the query.
```

<Callout type="info">
This command doesn't take into account orphan nodes, since the pattern we were doing was taking into account triplets. If you 
have in your dataset orphan nodes, consider using this command to create all the nodes prior to the triplet migration:

```cypher
CALL migrate.neo4j(
  "MATCH (n) RETURN labels(n) AS node_labels, elementId(n) as node_id, properties(n) as node_props", 
  {host: "localhost", port: 7687})
YIELD row 
MERGE (n:__MigrationNode__ {__elementId__: row.node_id})
SET n:row.node_labels
SET n += row.node_props
```
</Callout>

> ## Cleanup

Reviewer: ```suggestion
## Clean up temporary data
```

We actually don't need the `__MigrationNode__` label and the `__elementId__` property, so we will make sure to delete it from Memgraph:
```cypher
DROP INDEX ON :__MigrationNode__;
DROP INDEX ON :__MigrationNode__(__elementId__);
MATCH (n) SET n.__elementId__ = null;
```

> ## Create your own indices

Reviewer: ```suggestion
## Rebuild indices and constraints
```
